### PR TITLE
196 e2e testing

### PR DESCRIPTION
**Pull request by ethangardner**
Opening pull request to test GH action. Not ready for review yet.

### CODE REVIEW

**Comment by ethangardner** [comment]
@danielnaab - I think this is a good integration point if you'd like to review. I tried checking the e2e tests against local storage. That worked locally, but when it ran in CI/CD and hit the cloud.gov site, CI/CD was throwing for the `window.localStorage` call. I ended up changing it to just do source order in the DOM. 

Also, I added husky to the project to do precommit checks. I could add a step to do the docker build and e2e tests in that hook, but wanted to get your thoughts before I did. I'm trying not to have the commit process take too long so people just don't do `--no-verify` to bypass all the time.

**Comment by danielnaab** [comment]
@ethangardner I didn't get a chance to run locally yet. Nice how quickly the tests run in CI! I'm thinking it may be nice to have the e2e dependencies isolated in a separate package - ie, put a package.json in `./e2e` and move a few things into that directory. Thoughts?

File: apps/spotlight/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "astro": "astro",
     "build": "astro build",
     "clean": "rm -rf dist",
-    "dev": "astro dev",
+    "dev": "astro dev --host",
```
Comment (danielnaab) [inline apps/spotlight/package.json:9]: IIRC `pnpm` will also proxy args, so you could do `pnpm dev --host` in `Dockerfile.e2e`.

File: e2e/create.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+const uuidPattern = '[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}';
+
+const createNewForm = async (page: Page) => {
+  await page.goto(`${BASE_URL}/${GuidedFormCreation.getUrl()}`);
+  await page.getByRole('button', { name: 'Create New' }).click();
+}
+
+test('Create form from scratch', async ({ page }) => {
+  const createPage = new RegExp(`${uuidPattern}/${Create.slug}$`, 'i');
```
Comment (danielnaab) [inline e2e/create.spec.ts:14]: I don't think we'd actually want to add a `react-router-dom` dependency to this module, but it has a `generatePath` function that could be used in lieu of adding the slug - https://reactrouter.com/en/main/utils/generate-path

On 10x ASAP, we used https://www.npmjs.com/package/path-to-regexp as a lightweight option.

Just thinking out loud...

**Pull request merged**